microcosm_flask/formatting/base.py

```diff
@@ -1,17 +1,34 @@
+"""
+Response formatting base class.
+
+"""
 from abc import ABCMeta, abstractmethod
 
+from flask import Response
+from werkzeug.utils import get_content_type
+
 
 class BaseFormatter(metaclass=ABCMeta):
-    def __init__(self, response_schema=None):
-        # Formatting could need the response schema
-        # e.g. to specify column ordering in CSV response
-        self.response_schema = response_schema
 
-    @abstractmethod
-    def make_response(self, response_data, headers):
-        """
-        Build network response object from response data `dict`
-        The user can also pass in additional headers
+    def __call__(self, response_data, headers=None, **kwargs):
+        response = self.build_response(response_data)
+        headers = self.build_headers(headers=headers or {}, **kwargs)
+        response.headers.extend(headers)
```

Comment:
any point worrying about duplicate headers?

Reply:
Not really.